hack/jenkins/e2e.sh

```diff
@@ -565,6 +571,7 @@ case ${JOB_NAME} in
     : ${FAIL_ON_GCP_RESOURCE_LEAK:="true"}
     : ${GINKGO_TEST_ARGS:="--ginkgo.skip=$(join_regex_allow_empty \
           ${GKE_REQUIRED_SKIP_TESTS[@]:+${GKE_REQUIRED_SKIP_TESTS[@]}} \
+          "Daemon\sset|Deployment" \
```

Comment:
Added comment. Daemon sets and Deployments are experimental (not beta) features in the 1.1. release, and turned off for GKE.